src/main/java/com/google/sps/servlets/QuestionFormServlet.java

```diff
@@ -0,0 +1,123 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.common.flogger.FluentLogger;
+import java.io.IOException;
+import java.io.PrintWriter;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/** Servlet that creates a question form for the user to fill out.
+* A user can add a question to whichever test they want.
+* @author Klaudia Obieglo
+*/
+@WebServlet("/questionForm")
+public class QuestionFormServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  @Override
+  public void doGet(final HttpServletRequest request,
+        final HttpServletResponse response) throws IOException {
+    /*Returns all the questions that the user has created */
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      logger.atWarning().log("User is not logged in.");
+      response.sendError(HttpServletResponse.SC_UNAUTHORIZED,
+        "You are not authorised to view this page");
+      return;
+    }
+    logger.atInfo().log("User=%s is logged in", userService.getCurrentUser());
+    String ownerID = userService.getCurrentUser().getEmail();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    response.setContentType("text/html");
+    PrintWriter out = response.getWriter();
+    out.println("<!DOCTYPE html>");
```

Comment:
Try to avoid embedding frontend representation into your backend code. You can use a templating engine to decouple the frontend bits. E.g. in Python it's common to use https://en.wikipedia.org/wiki/Jinja_(template_engine), I'm sure Java has similar libraries.

Reply:
Issue created referencing this.